mixins/monorepo.mixin.tf

```diff
@@ -0,0 +1,63 @@
+# <-- BEGIN DOC -->
```

Comment:
rename to `remote-state.mixin.tf` ?

Reply:
I've renamed it to `infra-state.mixin.tf` as per our Slack discussion

Reply:
I think that `remote-state.mixin.tf` is confusing because we already have `remote-state.tf` in components. And `remote-state.mixin.tf` would suggest that it's the same thing except a mixin... except it's not. It's only applicable to repositories outside the monorepo. So I think the most suitable name (so far) is `infra-state.mixin.tf`.